### PR TITLE
[TR#19] Side Panel

### PR DESCRIPTION
**Pull request by Jeremy-Walton**
## Task

[TR #19](https://trello.com/c/RKaEKg10/19-side-panel)

## Why?

Side Panels are a common use-case for various application needs

## What Changed

- [X] Add Side Panel
- [X] Update Alert Docs JS to allow fixing a storybook documentation style conflict

## Sanity Check

- [X] Have you updated any usage of changed tokens?
- [X] Have you updated the docs with any component changes?
- [X] Have you updated the dependency graph with any component changes?
- [X] Have you run linters?
- [X] Have you run prettier?
- [X] Have you tried building the css?
- [X] Have you tried building storybook?
- ~~Do you need to update the package version?~~

## Screenshots

Screenshot 2023-07-26 at 10 29 39 AM
Screenshot 2023-07-26 at 10 29 57 AM
Screenshot 2023-07-26 at 10 30 03 AM
Screenshot 2023-07-26 at 10 31 05 AM
Screenshot 2023-07-26 at 10 31 10 AM
Screenshot 2023-07-26 at 10 31 19 AM
Screenshot 2023-07-26 at 10 31 22 AM
Screenshot 2023-07-26 at 10 31 26 AM
Screenshot 2023-07-26 at 10 31 30 AM
Screenshot 2023-07-26 at 10 31 37 AM

Screenshot 2023-07-27 at 1 07 45 PM
Screenshot 2023-07-27 at 1 07 50 PM

### CODE REVIEW

**Comment by Jeremy-Walton** [comment]
[Side Panel](https://trello.com/c/RKaEKg10/19-side-panel)

**Review by dallasbpeters — APPROVED** [review]
LGTM! :shipit:

File: src/components/side_panel.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  --_op-side-panel-width: calc(9 * var(--op-space-x-large));
+  --_op-side-panel-header-padding: var(--op-space-medium);
+  --_op-side-panel-body-padding: var(--op-space-medium);
+  --_op-side-panel-footer-padding: var(--op-space-medium);
+  --_op-side-panel-section-padding: var(--op-space-medium);
+
+  display: flex;
+  width: var(--_op-side-panel-width);
+  min-width: var(--_op-side-panel-width);
+  height: 100vh;
```
Comment (Jeremy-Walton) [inline src/components/side_panel.scss:12]: `height: 100%; // The 100vh is pushing things off the page.`

This assumes the panel is in something that extends fully to the bottom like a flex grow area.

**Comment by Jeremy-Walton** [comment]
[Build common layout templates in Figma with mirrored code in our docs](https://trello.com/c/Vuoq8eIQ/69-build-common-layout-templates-in-figma-with-mirrored-code-in-our-docs)

**Review by scriswell — APPROVED** [review]

**Pull request merged**